sm/plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -84,10 +87,18 @@ fn run(operation: PluginOp) -> Result<std::process::ExitStatus, InternalError> {
             }
         }
 
-        PluginOp::Remove {
-            module,
-            version: _unused,
-        } => run_cmd("apt-get", &format!("remove --quiet --yes {}", module))?,
+        PluginOp::Remove { module, version } => {
+            if let Some(version) = version {
+                // check the version mentioned present or not
+                if check_if_the_module_with_version_installed(&module, &version)? {
+                    run_cmd("apt-get", &format!("remove --quiet --yes {}", module))?
```

Comment:
I'm not happy with this behaviour as it will also deinstall packages that depend on the one that is to be removed. E.g. deinstallation of libreadline7 would deinstall more than 400 packages that depend on it. But this ticket here is probably not the right place to discuss this.

Reply:
Sure, file a bug.

Reply:
now its https://cumulocity.atlassian.net/browse/CIT-518